backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/MeetingService.cs

```diff
@@ -281,30 +314,47 @@ private MeetingEmailTemplateDto GenerateEmailParameters(Meeting meeting, string
 
         private EmailDto GenerateNewMeetingEmailTemplate(MeetingEmailTemplateDto parameters)
         {
-            return new EmailDto()
+            return new EmailDto
             {
                 Subject = $"New Event: {parameters.MeetingName} - {parameters.StartTime} - {parameters.LocationType}",
                 Body = $"Hi, {parameters.MemberName}!\n\n" +
-                $"A new event has been scheduled.\n" +
+                "A new event has been scheduled.\n" +
                 $"Event Type: {parameters.LocationType}\n" +
                 $"Invitee: {parameters.AuthorName}\n" +
                 $"Invitee Email: {parameters.AuthorEmail}\n" +
                 $"Event Date/Time: {parameters.StartTime}\n\n" +
                 $"View event in Easymeets: {parameters.Uri}{parameters.MeetingLink}"
             };
         }
+        
+        private EmailDto GenerateCancelledMeetingTemplate(MeetingEmailTemplateDto parameters)
+        {
+            return new EmailDto
+            {
+                Subject = $"Cancelled: {parameters.LocationType} with {parameters.AuthorName} on {parameters.StartTime.Date}",
+                Body = $"Hi, {parameters.MemberName},\n\n" +
+                "The event below has been canceled.\n" +
+                $"Invitee: {parameters.AuthorName}\n" +
+                $"Invitee Email: {parameters.AuthorEmail}\n" +
+                $"Event Date/Time: {parameters.StartTime}\n\n" +
+                $"Cancelled by: {parameters.AuthorName}"
+            };
+        }
 
-        private EmailDto GenerateEmailTemplate(Meeting meeting, string invitee, EmailTemplate? emailTemplate)
+        private EmailDto GenerateEmailTemplate(Meeting meeting, string invitee, TemplateType type, EmailTemplate? emailTemplate)
         {
             var emailParameters = GenerateEmailParameters(meeting, invitee);
             if (emailTemplate is not null)
             {
                 return _mapper.Map<EmailDto>(emailTemplate);
             }
-            else
+
+            return type switch
             {
-                return GenerateNewMeetingEmailTemplate(emailParameters);
-            }
+                TemplateType.Confirmation => GenerateNewMeetingEmailTemplate(emailParameters),
+                TemplateType.Cancellation => GenerateCancelledMeetingTemplate(emailParameters),
+                _ => new EmailDto()
```

Comment:
why we need here this "_" ?

Reply:
because without "_" we will have warning: "The switch expression does not handle some values of its input type"